src/common/freeze/deepfreeze.ts

```diff
@@ -0,0 +1,21 @@
+
+export const deepFreezeSkipDate = (o) => {
```

Comment:
That's a good point, there are a lot of issues with dates, we have been searching and not straight conclusion, in some places you can that deepfreeze cannot be applied to dates in some other the other way around. Maybe a solution is to store it as you mention, and then use selectors to convert them to date or string date.